imap/smtpclient.c

```diff
@@ -394,12 +394,25 @@ static int smtpclient_rset(smtpclient_t *sm)
     return r;
 }
 
-static int smtpclient_schk(smtpclient_t *sm)
+static int smtpclient_schk(smtpclient_t *sm, json_t *fromaddr)
```

Comment:
Is this the first time we pull in json into smtplib? If so, could we reuse the existing smtp_addr_t type (or a list of it) instead of a JMAP EmailAddress object?

Reply:
I'm not too keen on passing json into smtpclient.c either but I also didn't want to have to translate the json array into some other format, but its probably best that I do.

Reply:
Changed to using a strarray_t